astroid/nodes/node_classes.py

```diff
@@ -4567,15 +4650,10 @@ def postinit(
         self.patterns = patterns
         self.rest = rest
 
-    assigned_stmts: Callable[
-        [
-            "MatchMapping",
-            AssignName,
-            Optional[InferenceContext],
-            Literal[None],
-        ],
-        Generator[NodeNG, None, None],
-    ]
```

Comment:
Hmm, technically this and the once for `MatchClass` and `MatchAs` are more concrete than `AssignedStmtsCall`. I would suggest to leave them as is for now until we come back and refine (especially the return type) for `AssignedStmtsCall`.